ports/windows/README.md

```diff
@@ -3,46 +3,73 @@ It is based on Unix port, and expected to remain so.
 The port requires additional testing, debugging, and patches. Please
 consider to contribute.
 
+All gcc-based builds use the gcc compiler from [Mingw-w64](mingw-w64.org),
+which is the advancement of the original mingw project. The latter is
+getting obsolete and is not actively supported by MicroPython.
+
+Build instruction assume you're in the ports/windows directory.
 
 Building on Debian/Ubuntu Linux system
 ---------------------------------------
 
-    sudo apt-get install gcc-mingw-w64
+    sudo apt-get install python3 build-essential gcc-mingw-w64
+    make -C ../../mpy-cross CROSS_COMPILE=i686-w64-mingw32-
     make CROSS_COMPILE=i686-w64-mingw32-
 
-If for some reason the mingw-w64 crosscompiler is not available, you can try
-mingw32 instead, but it comes with a really old gcc which may produce some
-spurious errors (you may need to disable -Werror):
-
-    sudo apt-get install mingw32 mingw32-binutils mingw32-runtime
-    make CROSS_COMPILE=i586-mingw32msvc-
-
 
 Building under Cygwin
 ---------------------
 
-Install following packages using cygwin's setup.exe:
+Install Cygwin, then install following packages using Cygwin's setup.exe:
 
 * mingw64-i686-gcc-core
 * mingw64-x86_64-gcc-core
 * make
 
+Also install the python3 package, or install Python globally for Windows (see below).
+
 Build using:
 
+    make -C ../../mpy-cross CROSS_COMPILE=i686-w64-mingw32-
```

Comment:
Same problem as above?

Reply:
No, this gcc generates executables which run on windows and as such can also run when called from an MSYS2 shell, or Cygwin shell